### PR TITLE
simpleWindows with Visual Studio 2019 - New Azure4Students

### PR DESCRIPTION
**Pull request by Krisnatagoras**
### Best Practice Checklist
Check these items before submitting a PR... See the Contribution Guide for the full detail: https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/README.md 

1. uri's compatible with all clouds (Stack, China, Government)
1. Staged artifacts use _artifactsLocation & _artifactsLocationSasToken
1. Use a parameter for resource locations with the defaultValue set to resourceGroup().location
1. Folder names for artifacts (nestedtemplates, scripts, DSC)
1. Use literal values for apiVersion (no variables)
1. Parameter files (GEN-UNIQUE for value generation and no "changemeplease" values)
1. $schema and other uris use https
1. Use uniqueString() whenever possible to generate names for resources.  While this is not required, it's one of the most common failure points in a deployment. 
1. Update the metadata.json with the current date

For details: https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/best-practices.md

- [ ] - Please check this box once you've submitted the PR if you've read through the Contribution Guide and best practices checklist.

### Changelog

*
*
*

### CODE REVIEW

File: 101-vm-simple-windows-visualstudio2019/azuredeploy.json [hunk 1]
```diff
@@ -0,0 +1,259 @@
...
+        "vmName": {
+            "type": "string",
+            "defaultValue": "[concat('vm', uniqueString(resourceGroup().name))]",
+            "metadata": {
+                "description": "The name of you Virtual Machine."
+            }
+        },
+        "VmSize": {
+            "type": "string",
+            "allowedValues": [
```
Comment (bmoore-msft) [inline 101-vm-simple-windows-visualstudio2019/azuredeploy.json]: why are we limiting to these VM sizes?

File: 101-vm-simple-windows-visualstudio2019/azuredeploy.json [hunk 2]
```diff
@@ -0,0 +1,259 @@
...
+                "Standard_D2_v2",
+                "Standard_D3_v2",
+                "Standard_D1_v2",
+                "Standard_D4_v2",
+                "Standard_D5_v2"
+        ],
+            "metadata": {
+                "description": "The size of the VM"
+            },
+        "defaultValue": "Standard_D2_v2"
```
Comment (bmoore-msft) [inline 101-vm-simple-windows-visualstudio2019/azuredeploy.json]: move this above allowedValues

File: 101-vm-simple-windows-visualstudio2019/azuredeploy.json [hunk 3]
```diff
@@ -0,0 +1,259 @@
...
+            "defaultValue": "new",
+            "allowedValues": [
+                "new",
+                "existing"
+            ],
+            "metadata": {
+                "description": "Specify whether to create a new or existing NSG and vNet."
+            }
+        },
+        "virtualNetworkName": {
```
Comment (bmoore-msft) [inline 101-vm-simple-windows-visualstudio2019/azuredeploy.json]: need to include the vnet resourceGroup name for the existing vnet scenario

File: 101-vm-simple-windows-visualstudio2019/azuredeploy.json [hunk 4]
```diff
@@ -0,0 +1,259 @@
...
+        "location": {
+            "type": "string",
+            "defaultValue": "[resourceGroup().location]",
+            "metadata": {
+                    "description": "Location for all resources."
+            }
+        },
+        "dnsLabelPrefix": {
+            "type": "string",
+            "defaultValue": "[toLower(concat('dns-',parameters('vmName')))]",
```
Comment (bmoore-msft) [inline 101-vm-simple-windows-visualstudio2019/azuredeploy.json]: use a globally unique default

File: 101-vm-simple-windows-visualstudio2019/azuredeploy.parameters.json [hunk 5]
```diff
@@ -0,0 +1,15 @@
...
+    "$schema": "https://schema.management.azure.com/schemas/2015-01-01/deploymentParameters.json#",
+    "contentVersion": "1.0.0.0",
+    "parameters": {
+        "adminUsername": {
+            "value": "GEN-UNIQUE"
+        },
+        "adminPassword": {
+            "value": "GEN-PASSWORD"
+        },
+        "vmName": {
```
Comment (bmoore-msft) [inline 101-vm-simple-windows-visualstudio2019/azuredeploy.parameters.json]: remove this and use the defaultValue you have in the template

File: 101-vm-simple-windows-visualstudio2019/azuredeploy.json (hunk 1 above)
Comment (leestott) [reply 101-vm-simple-windows-visualstudio2019/azuredeploy.json]: @bmoore-msft limiting size to a VM we know will work as all A series VMs have been depreciated for Azure for Student so we are selecting a base machine which can be installed and scaled if required via the Portal

Comment (bmoore-msft) [reply 101-vm-simple-windows-visualstudio2019/azuredeploy.json]: I think this is going to be too limiting - we're excluding new sizes, "S" sizes and it won't account for quota, capacity, policy restrictions.  I think we need to leave this unbounded and can add appropriate documentation to the gettingstarted.md

File: 101-vm-simple-windows-visualstudio2019/azuredeploy.json (hunk 2 above)
Comment (Krisnatagoras) [reply 101-vm-simple-windows-visualstudio2019/azuredeploy.json]: Sure thing! Done!

File: 101-vm-simple-windows-visualstudio2019/azuredeploy.json (hunk 4 above)
Comment (Krisnatagoras) [reply 101-vm-simple-windows-visualstudio2019/azuredeploy.json]: Now has this value:
"defaultValue": "[toLower(concat('dns-', uniqueString(parameters('vmName'))))]"

File: 101-vm-simple-windows-visualstudio2019/azuredeploy.parameters.json (hunk 5 above)
Comment (Krisnatagoras) [reply 101-vm-simple-windows-visualstudio2019/azuredeploy.parameters.json]: Done!

File: 101-vm-simple-windows-visualstudio2019/azuredeploy.json (hunk 4 above)
Comment (bmoore-msft) [reply 101-vm-simple-windows-visualstudio2019/azuredeploy.json]: use something descriptive of the vm for your prefix (dns is too generic)

Comment (Krisnatagoras) [reply 101-vm-simple-windows-visualstudio2019/azuredeploy.json]: Sorry for not be so creative...

I came with this:

"defaultValue": "[toLower(concat('fqdn-', parameters('vmName')))]",

Is it better?

File: 101-vm-simple-windows-visualstudio2019/azuredeploy.json (hunk 3 above)
Comment (Krisnatagoras) [reply 101-vm-simple-windows-visualstudio2019/azuredeploy.json]: Do you mean on concat the vNet with RG name?

**Pull request merged**